circuits/src/zk_circuits/valid_match_settle/multi_prover.rs

```diff
@@ -230,16 +241,185 @@ impl ValidMatchSettle {
 
 // --- Settlement --- //
 
-impl ValidMatchSettle {
+impl<const MAX_BALANCES: usize, const MAX_ORDERS: usize, const MAX_FEES: usize>
+    ValidMatchSettle<MAX_BALANCES, MAX_ORDERS, MAX_FEES>
+where
+    [(); MAX_BALANCES + MAX_ORDERS + MAX_FEES]: Sized,
+{
     /// Validate settlement of a match result into the wallets of the two
     /// parties
-    #[allow(clippy::needless_pass_by_ref_mut)]
     pub(crate) fn validate_settlement(
-        witness: &ValidMatchSettleWitnessVar,
-        statement: (),
+        witness: &ValidMatchSettleWitnessVar<MAX_BALANCES, MAX_ORDERS, MAX_FEES>,
```

Comment:
this is copied from valid settle directly and refactored for use in `mpc-plonk`, same with the single prover impl